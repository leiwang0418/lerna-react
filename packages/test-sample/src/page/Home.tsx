import React, { FC } from 'react';
import PropTypes from 'prop-types';
import { BrowserRouter as Router, Switch, Route } from 'react-router-dom';
import UserSelection from './UserSelection';
import PublicRepositoriesList from './PublicRepositoriesList';

interface Props {
	isEditingUsername?: boolean;
}

const Home: FC<Props> = ({ isEditingUsername }) => (
	<Router>
		<Switch>
			{isEditingUsername && <Route path="/" component={UserSelection} />}
			<Route path="/" component={PublicRepositoriesList} />
		</Switch>
	</Router>
);

Home.defaultProps = {
	isEditingUsername: false
};

Home.propTypes = {
	isEditingUsername: PropTypes.bool
};

export default Home;
