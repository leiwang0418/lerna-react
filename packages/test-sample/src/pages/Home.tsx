import React, { FC } from 'react';
import { BrowserRouter as Router, Switch, Route } from 'react-router-dom';
import UserSelection from './UserSelection';
import PublicRepositoriesList from './PublicRepositoriesList';
import { useAppSelector } from '../store/hooks';
import { selectIsEditing } from './HomeSlice';

const Home: FC = () => {
	const isEditing = useAppSelector(selectIsEditing);
	return (
		<Router>
			<Switch>
				{isEditing && (
					<Route path="/" component={UserSelection} />
				)}
				<Route path="/" component={PublicRepositoriesList} />
			</Switch>
		</Router>
	);
};

export default Home;
