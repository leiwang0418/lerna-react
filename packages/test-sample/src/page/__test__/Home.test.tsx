import React from 'react';
import { render, screen } from '@testing-library/react';
import Home from '../Home';
import MockUserSelection from '../UserSelection';
import MockPublicRepositoriesList from '../PublicRepositoriesList';


jest.mock('../UserSelection', () => () => <div />);
jest.mock('../PublicRepositoriesList', () => () => <div />);

describe('snapshot test with mock', () => {
	test('renders the basic home', () => {
		const { container } = render(<Home />);

		expect(container).toMatchSnapshot();
	});
});
