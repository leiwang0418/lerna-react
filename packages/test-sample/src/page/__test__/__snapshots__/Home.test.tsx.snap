// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test with mock renders the basic home 1`] = `
<div>
  <div />
</div>
`;
